src/generate/generic.rs

```diff
@@ -227,3 +227,62 @@ pub enum Variant<U, T> {
     ///Raw bits
     Res(U),
 }
+
+use Variant::*;
+impl<U, T> Variant<U, T> {
+    /// Check if the variant is expected
+    pub fn is_value(&self) -> bool {
```

Comment:
This naming is strange. At least function name does not correspond to the docs. I can't suggest anything nicer, though.

Reply:
Do you have suggestions?